apps/verticalhome/js/app.js

```diff
@@ -126,10 +132,19 @@
             window.dispatchEvent(new CustomEvent('moz-app-loaded'));
           });
       }.bind(this));
+
+      this.pinManager = new PinAppManager();
+      navigator.mozL10n.ready(this.l10nInit.bind(this));
```

Comment:
Since this is all related to the time format, it would be better to put it in the clock module.

Also note that .ready() only gets called once. You probably want to be notified every time the locale changes, so that the clock format can change dynamically if the user switches locales.